core/src/main/java/io/parsingdata/metal/expression/value/reference/Last.java

```diff
@@ -49,4 +50,16 @@ public String toString() {
         return getClass().getSimpleName() + "(" + operand + ")";
     }
 
+    @Override
+    public boolean equals(Object obj) {
+        return obj != null
+            && getClass() == obj.getClass()
+            && Objects.equals(operand, ((Last)obj).operand);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(operand);
```

Comment:
It seems that `getClass()` is cheap, so that might be a good idea. We could even cache it statically. I've made an issue: #150.